devel/libwebsockets/Portfile

```diff
@@ -0,0 +1,34 @@
+# -*- coding: utf-8; mode: tcl; indent-tabs-mode: nil -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           github 1.0
+PortGroup           cmake 1.1
+
+github.setup        warmcat libwebsockets 2.4.1 v
+categories          devel net
+platforms           darwin
+license             LPGL-2.1
+
+maintainers         nomaintainer
+
+description         Canonical libwebsockets.org websocket library
+
+long_description    \
+    Libwebsockets is a lightweight pure C library built to use minimal \
+    CPU and memory resources, and provide fast throughput in both directions \
+    as client or server.
+
+distname            v${version}
+worksrcdir           ${name}-${version}
+
+homepage            https://github.com/warmcat/libwebsockets
+master_sites        https://github.com/warmcat/libwebsockets/archive
```

Comment:
All those variables (`distname`, `worksrcdir`, `homepage`, `master_sites`) should work automatically with the github portgroup.

See https://github.com/macports/macports-guide/pull/12 ;)

Reply:
It  should, but their GitHub archive isn't standard, evidently - e.g., archive is named v2.4.1.tar.gz, but extracts to libwebsockets-2.4.1, etc.  This was smallest change that I could think of to make it work :)

Reply:
Someone should probably take a look then. There are a couple of tricks to make this stuff work automatically and we should try to use them (if it's not possible, that's fine, but if it is, we should strive for simplicity).

Reply:
Hi Mojca,
I'm late in replying, and now I can't find your comment about becoming maintainer, so I'm replying here.  To refresh, you asked if I'd be willing to maintainer two Port submissions, which I agreed to.  Finally got to read MacPorts Guide and updated Port maintainers.  I used the syntax Ryan suggested in another email:

Reply:
maintainers  {slewsys @slewsys} openmaintainer

But while I have an account on trac.macports.org, I don't appear to have a macports email address:  email to slewsys@macports.org bounces.  So in your message that I can't find, you asked that I provide an obfuscated email address.   If I understood Ryan's email, that address should not be used in the Port itself, so I'll provide it here for the sake of completeness: gmail.com:slewsys.